prow/flagutil/github.go

```diff
@@ -40,6 +44,24 @@ func (o *GitHubOptions) AddFlags(fs *flag.FlagSet) {
 	fs.Var(&o.endpoint, "github-endpoint", "GitHub's API endpoint (may differ for enterprise).")
 	fs.StringVar(&o.TokenPath, "github-token-path", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
 	fs.StringVar(&o.deprecatedTokenFile, "github-token-file", "", "DEPRECATED: use -github-token-path instead.  -github-token-file may be removed anytime after 2019-01-01.")
+	o.baseURL = getBaseURL(o.endpoint)
+
+}
+
+// getBaseURL infer base url from the github endpoint
+func getBaseURL(endpoints Strings) string {
+	for _, endpoint := range endpoints.vals {
+		endpointURL, err := url.Parse(endpoint)
+		if err != nil {
+			continue
+		}
+		if endpointURL.Host != "api.github.com" {
+			return strings.Join([]string{endpointURL.Scheme, endpointURL.Host}, "://")
```

Comment:
This will output the endpoint api which is different from baseURL, as example:

- base url: `https://github.mycorp.com`
- endpoint api: `https://github.mycorp.com/apis/v3`